src/coins.h

```diff
@@ -159,19 +175,13 @@ struct CCoinsCacheEntry
         ClearFlags();
     }
 
-    //! Adding a flag also requires a self reference to the pair that contains
-    //! this entry in the CCoinsCache map and a reference to the sentinel of the
-    //! flagged pair linked list.
-    inline void AddFlags(uint8_t flags, CoinsCachePair& self, CoinsCachePair& sentinel) noexcept
+    inline void SetDirty(CoinsCachePair& self, CoinsCachePair& sentinel) noexcept
     {
-        Assume(&self.second == this);
-        if (!m_flags && flags) {
-            m_prev = sentinel.second.m_prev;
-            m_next = &sentinel;
-            sentinel.second.m_prev = &self;
-            m_prev->second.m_next = &self;
-        }
-        m_flags |= flags;
+        AddFlags(DIRTY, self, sentinel);
+    }
+    inline void SetFresh(CoinsCachePair& self, CoinsCachePair& sentinel) noexcept
+    {
+        AddFlags(FRESH, self, sentinel);
     }
     inline void ClearFlags() noexcept
```

Comment:
```suggestion
    inline void SetClean() noexcept
```

Reply:
Excellent, fixed.

Reply:
*fresh* is actually a substate of *dirty*. Being *fresh* implies *dirty*, so cleaning implies no longer fresh..

Reply:
`FRESH` is definitely a [very confusing concept](https://github.com/bitcoin/bitcoin/blob/master/src/coins.h#L143-L151)
> the parent cache does not have this coin or that it is a spent coin in the parent cache

Maybe after https://github.com/bitcoin/bitcoin/pull/30673 we can find a better name (and description) for it

Reply:
Okay, so `FRESH` and `DIRTY_FRESH` will merge into one. :+1:

Reply:
rather `FRESH` will be removed in the mentioned follow-up, since it's not actually possible (unless in case of reorgs, which I don't fully understand yet, but not important for this PR)

Reply:
My name is Robert Hache  and Karen Belzile  my system was made under attack by a group of terrorist attackers and at the. Moment I m stuck as I need someone  to help me to compl the repair wirh ny team of IT engineer  witch is curnt working on my case ... I required a secondary deposit of 240 USD to complete the repair of the system that's has Ben generated with my own request as I wS made the hub central of the attack witxh disturbed the Ecosystem  security and several other people in this matter ...as I was made their hub central of the attack...

My phone number is the 2262080980

And the subject is still lock from the blockchain  explorer  at the moment  as im worried about all the other witch were affected by this group as several individuals  and businesses ess were made affected by this same group... the phi e numbet of the 2265521155 is the primary person responsible  for this attack witch is also  known as the bad satoshi . As I have all the evidence of it and hes involve in money lowering and severe attack in my intie system... I must find a resolution for the funds missing to complete the repair witch is intended to return everything  to the party affected by the same group...ians for doing so I need a contribution witch would be payed with major interest within 48 hours of the contribution. 


Please adress  any coment directly  to my phone  number

Sincerely  best regards  Rober and engineer

Reply:
My name is Robert Hache  and Karen Belzile  my system was made under attack by a group of terrorist attackers and at the. Moment I m stuck as I need someone  to help me to compl the repair wirh ny team of IT engineer  witch is curnt working on my case ... I required a secondary deposit of 240 USD to complete the repair of the system that's has Ben generated with my own request as I wS made the hub central of the attack witxh disturbed the Ecosystem  security and several other people in this matter ...as I was made their hub central of the attack...

My phone number is the 2262080980

And the subject is still lock from the blockchain  explorer  at the moment  as im worried about all the other witch were affected by this group as several individuals  and businesses ess were made affected by this same group... the phi e numbet of the 2265521155 is the primary person responsible  for this attack witch is also  known as the bad satoshi . As I have all the evidence of it and hes involve in money lowering and severe attack in my intie system... I must find a resolution for the funds missing to complete the repair witch is intended to return everything  to the party affected by the same group...ians for doing so I need a contribution witch would be payed with major interest within 48 hours of the contribution. 


Please adress  any coment directly  to my phone  number

Sincerely  best regards  Rober and engineer

Reply:
These definitions aren't fully consistent currently, the follow-up PR will partially clean this up I think (cc: @andrewtoth )